### ISSUE TITLE
Wrong calculation of .container variables

### ISSUE THREAD

**Issue by franz-josef-kaiser** [twbs/bootstrap#15623]
I was looking at the screen sizes in the `~/less/variables.less` file:

```
// Small screen / tablet
@container-tablet:             (720px + @grid-gutter-width);
//** For `@screen-sm-min` and up.
@container-sm:                 @container-tablet;

// Medium screen / desktop
@container-desktop:            (940px + @grid-gutter-width);
//** For `@screen-md-min` and up.
@container-md:                 @container-desktop;

// Large screen / wide desktop
@container-large-desktop:      (1140px + @grid-gutter-width);
//** For `@screen-lg-min` and up.
@container-lg:                 @container-large-desktop;
```

When looking at how the `@media` query breakpoints are calculated:

```
// Extra small screen / phone
//** Deprecated `@screen-xs` as of v3.0.1
@screen-xs:                  480px;
//** Deprecated `@screen-xs-min` as of v3.2.0
@screen-xs-min:              @screen-xs;
//** Deprecated `@screen-phone` as of v3.0.1
@screen-phone:               @screen-xs-min;

// Small screen / tablet
//** Deprecated `@screen-sm` as of v3.0.1
@screen-sm:                  768px;
@screen-sm-min:              @screen-sm;
//** Deprecated `@screen-tablet` as of v3.0.1
@screen-tablet:              @screen-sm-min;

// Medium screen / desktop
//** Deprecated `@screen-md` as of v3.0.1
@screen-md:                  992px;
@screen-md-min:              @screen-md;
//** Deprecated `@screen-desktop` as of v3.0.1
@screen-desktop:             @screen-md-min;

// Large screen / wide desktop
//** Deprecated `@screen-lg` as of v3.0.1
@screen-lg:                  1200px;
@screen-lg-min:              @screen-lg;
//** Deprecated `@screen-lg-desktop` as of v3.0.1
@screen-lg-desktop:          @screen-lg-min;

// So media queries don't overlap when required, provide a maximum
@screen-xs-max:              (@screen-sm-min - 1);
@screen-sm-max:              (@screen-md-min - 1);
@screen-md-max:              (@screen-lg-min - 1);
```

then it seems to be quite off to assume a fixed `@grid-gutter-width` in the calculation. 

    @grid-gutter-width:         30px;

Yes, this math is "right" when we got exactly the `30px`:

    2 x 30px = 60px + 1140px = @container-large-desktop

But if the `@grid-gutter-width` changes ...

In short: The math is simply done from the wrong side: The right calculation would be:

    @container-large-desktop: (@screen-lg - @grid-gutter-width);

...unless I got something completely wrong here (don't do much with TWBS).

---

Current version: 3.2.x

**Comment by mdo** [comment]
There is no specific formula for calculating the media breakpoints—those are based on the viewport sizes on specific device ranges. We then use our best judgement and experience to determine what could be a maximum container size for those environments.

Long story short, the correlation is not as direct or 1:1 as you might expect.

**Comment by bassjobsen** [comment]
> There is no specific formula for calculating the media breakpoints

The above also means that there is no relation between the break points (@screen-*) and the container size (@container-*). So instead of `@container-tablet:             (720px + @grid-gutter-width);` you can possible use:

    @container-*: floor( ( @screen-* - @grid-gutter-width ) / @grid-columns ) * @grid-columns + @grid-gutter-width;

The above will generate with default setting:

    @container-sm: 762px;
    @container-md: 990px;
    @container-lg: 1194px;
